libass/ass_outline.c

```diff
@@ -244,6 +276,10 @@ bool outline_transform_2d(ASS_Outline *outline, const ASS_Outline *source,
         for (int k = 0; k < 2; k++)
             v[k] = m[k][0] * pt[i].x + m[k][1] * pt[i].y + m[k][2];
 
+        if (!(fabs(v[0]) <= OUTLINE_MAX && fabs(v[1]) <= OUTLINE_MAX)) {
```

Comment:
Nit-pick: When comparing with floating types OUTLINE_MAX will be cast to floating type as well — but an accurate representation may not necessarily exist. Checking `<=` instead of `<` to avoid overflow can thus be potentially unsafe.
After a quick glance it seems like all float-comparisons here are done with doubles. On probably most systems `double` is 64 bit and able to accurately represent the current value of `OUTLINE_MAX`. 

Still, imho it would be better to stick to `<` when floating types are involved to account for odd systems and potential future changes of `OUTLINE_MAX` *(or is this specific value required for some VSFilter-compatibility?)*.

With floats this would already be problem on my machine:
```C
printf("%d\n", ((float) (OUTLINE_MAX+1)) <= OUTLINE_MAX);  // 1
printf("%d\n", ((double)(OUTLINE_MAX+1)) <= OUTLINE_MAX);  // 0
```

*same applies to ass_outline.c:313 and ass_render.c:1443*

Reply:
Done.

> is this specific value required for some VSFilter-compatibility?

No, it's to avoid overflow in cubic spline splitting, `(p0 + 3 * p1 + 3 * p2 + p3) / 8`.